pkg/controller/jobframework/validation.go

```diff
@@ -83,3 +84,13 @@ func ValidateUpdateForParentWorkload(oldJob, newJob GenericJob) field.ErrorList
 	}
 	return allErrs
 }
+
+func ValidateUpdateForWorkloadPriorityClassName(oldJob, newJob GenericJob) field.ErrorList {
```

Comment:
Since `validation.go` doesn't have test file like `validation_test.go`, I added test cases to `job_webhook_test.go` etc instead. This is consistent with other validation functions like `ValidateUpdateForParentWorkload`.